errbot/backends/slack.py

```diff
@@ -565,6 +566,25 @@ def send_message(self, mess):
                 "to %s: %s" % (to_humanreadable, mess.body)
             )
 
+    def _slack_upload(self, stream):
+        """Perform upload defined in a stream."""
+        stream.accept()
```

Comment:
I have almost the same code in my branch, with logs working with no problem. The only thing is that failures are silent so you need protect the whole method with a try / catch Exception.